rocAL/include/readers/webdataset_source_reader.h

```diff
@@ -0,0 +1,145 @@
+    //!< _batch_count Defines the quantum count of the images to be read. It's
+    //!< usually equal to the user's batch size.
+    /// The loader will repeat images if necessary to be able to have images
+    /// available in multiples of the load_batch_count, for instance if there
+    /// are 10 images in the dataset and _batch_count is 3, the loader repeats 2
+    /// images as if there are 12 images available.
+    size_t _batch_count = 1;
+    size_t _file_id = 0;
+    size_t _in_batch_read_count = 0;
+    bool _loop;
+    bool _shuffle;
+    int _read_counter = 0;
+    void incremenet_read_ptr();
+    int release();
+    size_t get_file_shard_id();
+    void incremenet_file_id() { _file_id++; }
+    void parse_tar_files(std::vector<SampleDescription> &samples_container, std::vector<ComponentDescription> &components_container, std::unique_ptr<FileIOStream> &tar_file);
+    void parse_index_files(std::vector<SampleDescription> &samples_container,
+                      std::vector<ComponentDescription> &components_container,
+                      const std::string &index_path);
+    void parse_sample_description(
+        std::vector<SampleDescription> &samples_container,
+        std::vector<ComponentDescription> &components_container,
+        std::ifstream &index_file, const std::string &index_path, int64_t line,
+        int index_version);
+    Reader::Status webdataset_record_reader_from_components(ComponentDescription component, unsigned wds_shard_index);
+    std::shared_ptr<MetaDataReader> _meta_data_reader = nullptr;
+    std::vector<std::unique_ptr<FileIOStream>> _wds_shards;
+    Reader::Status read_web_dataset_at_offset(unsigned char *buff,
+                                              std::string file_name,
+                                              uint file_size, uint offset,
+                                              uint wds_shard_index);
+    void increment_curr_file_idx();
+    unsigned _shard_start_idx;
+    signed _shard_size = -1;
+    size_t _shard_id = 0;
+    size_t _shard_count = 1;  // equivalent of batch size
+    //!< _file_count_all_shards total_number of files in to figure out the max_batch_size (usually needed for distributed training).
+    size_t _file_count_all_shards;
+    size_t _padded_samples = 0;
+    std::pair<RocalBatchPolicy, bool> _last_batch_info;
+    size_t _last_batch_padded_size = 0;
+    size_t _num_padded_samples = 0;
+    bool _stick_to_shard = false;
+    bool _pad_last_batch_repeated = false;
+    size_t get_start_idx(); // Start Idx of the Shard's Data
+    size_t get_dataset_size(); // DataSet Size
+    size_t actual_shard_size_without_padding(); // Number of files belonging to a shard (without padding)
+    size_t largest_shard_size_without_padding(); // Number of files belonging to a shard (with padding)
+    //!< Used to advance to the next shard's data to increase the entropy of the data seen by the pipeline>
+    void increment_shard_id();
```

Comment:
All these have been removed in file source reader and other readers can you please check